### PR TITLE
Fix two taps required to interact with player when in fullscreen mode

### PR DESCRIPTION
**Pull request by anilbeesetti**
previously we had to tap twice to show controllers. On first tap the status bar will be shown and on the next tap player controller will be shown. This pr fixes this by showing controller on single tap

### CODE REVIEW

File: app/src/main/java/com/github/libretube/ui/extensions/Activity.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import android.app.Activity
+import androidx.core.view.WindowCompat
+import androidx.core.view.WindowInsetsCompat
+import androidx.core.view.WindowInsetsControllerCompat
+
+fun Activity.hideSystemBars() {
+    WindowCompat.getInsetsController(window, window.decorView).apply {
+        systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
+        hide(WindowInsetsCompat.Type.systemBars() or WindowInsetsCompat.Type.navigationBars())
```
Comment (Bnyro) [inline app/src/main/java/com/github/libretube/ui/extensions/Activity.kt:11]: The navigation bars shouldn't be made visible here as that's just a bad practice for UX. We also had some issues open about that previously.

**Review by Bnyro — CHANGES_REQUESTED** [review]

**Comment by Bnyro** [comment]
Also, please remove the unused imports in the `WindowHelper` file.

File: app/src/main/java/com/github/libretube/ui/extensions/Activity.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    }
+}
+
+fun Activity.showSystemBars() {
+    WindowCompat.getInsetsController(window, window.decorView).apply {
+        systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
```
Comment (Isira-Seneviratne) [inline app/src/main/java/com/github/libretube/ui/extensions/Activity.kt:17]:
```suggestion
        systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_BARS_BY_TOUCH
```

Comment (Isira-Seneviratne) [reply app/src/main/java/com/github/libretube/ui/extensions/Activity.kt:17]: Also, these two methods could be easily combined into one, e.g. `toggleSystemBars(show: Boolean)`.

Comment (anilbeesetti) [reply app/src/main/java/com/github/libretube/ui/extensions/Activity.kt:17]: No. `WindowInsetsControllerCompat.BEHAVIOR_SHOW_BARS_BY_TOUCH` conflict with touch to show player controller. i.e now we have to tap twice to show player controller

File: app/src/main/java/com/github/libretube/ui/views/CustomExoPlayerView.kt [hunk 3]
```diff
@@ -202,6 +205,22 @@ internal class CustomExoPlayerView(
...
+                playerViewModel?.isFullscreen?.value?.let { isFullscreen ->
+                    if (!isFullscreen) return@let
+                    when (visibility) {
+                        View.VISIBLE -> {
+                            activity.showSystemBars(WindowInsetsCompat.Type.statusBars())
+                        }
+                        View.GONE -> {
+                            activity.hideSystemBars(WindowInsetsCompat.Type.statusBars())
+                        }
+                    }
```
Comment (Isira-Seneviratne) [inline app/src/main/java/com/github/libretube/ui/views/CustomExoPlayerView.kt:219]:
```suggestion
                    val windowInsetsControllerCompat = WindowCompat.getInsetsController(activity.window, activity.window.decorView)
                    windowInsetsControllerCompat.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
                    if (visibility == View.VISIBLE) {
                        windowInsetsControllerCompat.show(WindowInsetsCompat.Type.statusBars())
                    } else {
                        windowInsetsControllerCompat.hide(WindowInsetsCompat.Type.statusBars())
                    }
```

Comment (anilbeesetti) [reply app/src/main/java/com/github/libretube/ui/views/CustomExoPlayerView.kt:219]: What's wrong with when

Comment (Isira-Seneviratne) [reply app/src/main/java/com/github/libretube/ui/views/CustomExoPlayerView.kt:219]: According to the documentation for the listener, only View.VISIBLE and View.GONE are passed, so this is simpler.

File: app/src/main/java/com/github/libretube/helpers/WindowHelper.kt [hunk 4]
```diff
@@ -20,25 +21,10 @@ object WindowHelper {
 
         WindowCompat.setDecorFitsSystemWindows(window, !isFullscreen)
 
-        val controller = WindowCompat.getInsetsController(window, window.decorView)
-        val flags = WindowInsetsCompat.Type.systemBars() or WindowInsetsCompat.Type.navigationBars()
         if (isFullscreen) {
-            controller.hide(flags)
+            activity.hideSystemBars(WindowInsetsCompat.Type.systemBars())
```
Comment (Bnyro) [inline app/src/main/java/com/github/libretube/helpers/WindowHelper.kt:25]: Due to #3069, hiding the navigation bars here too might fix issues for some devices.

File: app/src/main/java/com/github/libretube/ui/views/CustomExoPlayerView.kt [hunk 5]
```diff
@@ -202,6 +205,22 @@ internal class CustomExoPlayerView(
...
+        setControllerVisibilityListener(
+            ControllerVisibilityListener { visibility ->
+                playerViewModel?.isFullscreen?.value?.let { isFullscreen ->
+                    if (!isFullscreen) return@let
+                    when (visibility) {
+                        View.VISIBLE -> {
+                            activity.showSystemBars(WindowInsetsCompat.Type.statusBars())
+                        }
+                        View.GONE -> {
+                            activity.hideSystemBars(WindowInsetsCompat.Type.statusBars())
```
Comment (Bnyro) [inline app/src/main/java/com/github/libretube/ui/views/CustomExoPlayerView.kt:217]: Same as above: Due to #3069, hiding the navigation bars here too might fix issues for some devices.

**Comment by Bnyro** [comment]
I feel like we could also just create a function ```
fun Activity.toggleSystemBars(showBars: Boolean)
```
which then calls the corresponding methods since we're currently doing two boolean if expressions and different places which could be simplified that way. Though I'm not sure whether that's really needed since it's only being used twice in total, but I probably wouldn't be too bad.

File: app/src/main/java/com/github/libretube/helpers/WindowHelper.kt (hunk 4 above)
Comment (anilbeesetti) [reply app/src/main/java/com/github/libretube/helpers/WindowHelper.kt:25]: Screenshot_20230401-122249_MX_Player_Pro

Do want navigation bars this way or not appear when controls are visible

Comment (Bnyro) [reply app/src/main/java/com/github/libretube/helpers/WindowHelper.kt:25]: They should not appear, but if the user manually shows them they should be hidden again automatically.

**Comment by anilbeesetti** [comment]
see this video for reference

https://user-images.githubusercontent.com/66936649/229292694-205d9840-d41d-4f30-9df6-6b7ab8f33fbd.mp4

**Review by Bnyro — APPROVED** [review]

**Comment by Bnyro** [comment]
Thank you!

**Pull request merged**